src/System.IdentityModel.Tokens.Jwt/JwtPayload.cs

```diff
@@ -296,11 +328,10 @@ public IList<string> Aud
             {
                 if (_audiences == null)
                 {
-                    List<string> tmp = GetListOfClaims(JwtRegisteredClaimNames.Aud);
+                    HashSet<string> tmp = new (GetListOfClaims(JwtRegisteredClaimNames.Aud));
                     Interlocked.CompareExchange(ref _audiences, tmp, null);
                 }
-
-                return _audiences;
+                return [.. _audiences];
```

Comment:
This will create a new allocation every time

Reply:
only if _audiences is null though so it would only impact concurrent calls until one succeeds at setting _audiences, as soon as one succeeds no future calls will hit this block.

Reply:
fair, I wasn't sure... it won't be necessary anymore since we are getting rid of hashset

Reply:
We might still need to do deduplication depending on conversation with Brent. The hashset would require a new allocation every time which was the tradeoff for making deduplication O(n) we could do deduplication in place in a list but it will be less performant... What is preferred?

Reply:
Resolved offline, deduplication documented in code is for Aud claims defined in multiple places, not copies of the same Aud claim